src/lib/modernExtend.ts

```diff
@@ -116,10 +118,12 @@ export function setupConfigureForReporting(
 }
 
 export function setupConfigureForBinding(cluster: string | number, clusterType: 'input' | 'output', endpointNames?: string[]) {
-    const configure: Configure = async (device, coordinatorEndpoint) => {
+    const configure: Configure = async (device, coordinatorEndpoint, definition) => {
         if (endpointNames) {
-            const endpointsMap = new Map<string, boolean>(endpointNames.map((e) => [e, true]));
-            const endpoints = device.endpoints.filter((e) => endpointsMap.has(e.ID.toString()));
+            const definitionEndpoints = definition.endpoint(device);
+            const endpointIds = endpointNames.map((e) => definitionEndpoints[e]);
+            const endpointsMap = new Map<number, boolean>(endpointIds.map((e) => [e, true]));
+            const endpoints = device.endpoints.filter((e) => endpointsMap.has(e.ID));
```

Comment:
```suggestion
            const endpoints = device.endpoints.filter((e) => endpointIds.includes(e.ID));
```

And then you can remove `const endpointsMap = new Map<number, boolean>(endpointIds.map((e) => [e, true]));`

Reply:
Thanks!